update wording and add URL links for quicker navigation (#110)
flowforge/handbook: development/release.md

### Setup

 - Decide who will be Release Manager for this release. For Major/Minor releases this should be shared across the whole team to prevent it becoming a single point of failure. For Fix releases it can be the developer committing the fix.

>  - Create a Release check list issue (using template) on the flowforge/admin project to keep track of all components included in the release (e.g. https://github.com/flowforge/admin/issues/80)

Reviewer: ```suggestion
 - Create a Release checklist issue using the [Release Checklist template](https://github.com/flowforge/admin/issues/new?assignees=&labels=&template=release.md&title=Release%3A)
```

We can link straight to the 'new issue' page with the appropriate issue template applied.

Author: perfect
